### PR TITLE
fix: onboarding style alignment

### PR DESCRIPTION
**Pull request by edmonday**
# Description

Fixed several alignment issues on the onboarding pages. 

[M – Onboarding: Elements alignment problems](https://3.basecamp.com/3105655/buckets/36562350/todos/7266606901/)
[M – Onboarding: Elements alignment problems 2](https://3.basecamp.com/3105655/buckets/36562350/todos/7267301902/)
[M – Signup Page image is distorted](https://3.basecamp.com/3105655/buckets/36562350/todos/7230472906/)
[M – Onboarding: mobile view improvements](https://3.basecamp.com/3105655/buckets/36562350/todos/7266613342/)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

copilot:walkthrough

### CODE REVIEW

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx [hunk 1]
```diff
@@ -36,92 +34,78 @@ export function OnboardingDrawer(): ReactElement {
...
-              priority
-            />
-          </Box>
-        </>
-      )}
-
-      {templateId != null && (
-        <Box
+      </Stack>
+      {templateId !== undefined ? (
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx:65]: noticed that this file have both `templateId !== undefined` and `templateId != null`. Can we stick to null check, should allow us to check both null and undefined value. Unless we're sure that templateId will be undefined for good

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx [hunk 2]
```diff
@@ -36,92 +34,78 @@ export function OnboardingDrawer(): ReactElement {
...
+                }}
+              >
+                <OnboardingTemplateCard templateId={templateId} />
+              </Stack>
+            )}
+            <Box sx={{ pl: 4 }}>
+              {newAccountQuery === true && (
+                <OnboardingStepper variant="desktop" />
+              )}
+            </Box>
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx:100]: Wrap the box inside the check?

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingLandingDrawer/OnboardingLandingDrawer.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    expect(getAllByRole('img')[1].getAttribute('src')).toBe(
+      'landing-description.png'
+    )
+  })
+
+  it('should render landing image with the stepper', async () => {
+    mockUseRouter.mockReturnValue({
+      push,
+      query: { newAccount: 'true' }
+    } as unknown as NextRouter)
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingLandingDrawer/OnboardingLandingDrawer.spec.tsx:52]: If newAccount value is being passed down as prop. Do we need to to mock use router?

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingPageWrapper.tsx [hunk 4]
```diff
@@ -26,38 +26,43 @@ export function OnboardingPageWrapper({
...
     <Stack
       direction={{ xs: 'column', md: 'row' }}
       sx={{
-        overflow: 'hidden',
         height: viewportHeight ?? '100vh',
         backgroundColor: { xs: 'background.default', md: 'background.paper' }
       }}
     >
       <OnboardingDrawer />
+
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingPageWrapper.tsx:40]: no need for space, can easily be read

**Review by GeronimoJohn — APPROVED** [review]

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx [hunk 5]
```diff
@@ -21,7 +19,7 @@ export function OnboardingDrawer(): ReactElement {
       : router.query.redirect?.includes('newAccount')
 
   function getTemplateId(): string | undefined {
-    const url = router.query.redirect as string | undefined
+    const url = router.query.redirect as string
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx:22]: don't need typecast as line 23 is performing null | undefined check

**Review by Kneesal — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx (hunk 5 above)
Comment (edmonday) [reply apps/journeys-admin/src/components/OnboardingPageWrapper/OnboardingDrawer/OnboardingDrawer.tsx:22]: line 25 requires url to be of type string

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent

**Review by Kneesal — APPROVED** [review]

**Pull request merged**